core/cas-server-core-services-registry/src/main/java/org/apereo/cas/services/resource/ResourceNamingStrategy.java

```diff
@@ -0,0 +1,22 @@
+package org.apereo.cas.services.resource;
+
+import org.apereo.cas.services.RegisteredService;
+
+/**
+ * This is {@link ResourceNamingStrategy}.  Interface to provide naming strategy
+ * to resource based services.
+ *
+ * @author Travis Schmidt
+ * @since 5.3.0
+ */
+public interface ResourceNamingStrategy {
```

Comment:
You probably could add @Functional here.

Also, I think a rename here is in order since this interface is specific to the service registry business: how about we switch to `RegisteredServiceResourceNamingStrategy`? I dont imagine we'd be using this interface for rename strategies for a variety of things, right? Makes sense to me keep the interface and impl specific in names.